File: Assignment9/src/Repo/MessageRepo.js
const load = async (auth, id) => {
  const url = `https://cse118.com/api/v2/channel/${id}/message`;
  const headers = {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${auth}`,
  };

  const response = await fetch(url, {
    method: 'GET',
    headers: headers,
  });

  const json = await response.json();
  //  console.log(json)
  return json;
};

export {load};

const add = async (auth, id, content) => {
  console.log('adding message');
  const url = `https://cse118.com/api/v2/channel/${id}/message`;
  const headers = {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${auth}`,
  };

  const data = {content};
  const jsonData = JSON.stringify(data);

  const response = await fetch(url, {
    method: 'POST',
    headers: headers,
    body: jsonData,
  });

  const responseData = await response.json();
  return responseData;
};

export {add};

const Delete = async (auth, id) => {
  console.log('deleting message');
  const url = `https://cse118.com/api/v2/message/${id}`;
  const headers = {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${auth}`,
  };

  const response = await fetch(url, {
    method: 'DELETE',
    headers: headers,
  });

  // The Swift code does not use the response data, so we can omit the next two lines in JavaScript
  console.log('Here');
  const responseData = await response;
  console.log(responseData);
  return responseData;
};

export {Delete};

File: Assignment6-1.Submission/Assignment6/Repo/MessageRepo.swift
//
//  WorkspaceRepo.swift
//  Assignment6
//
//  Created by Abhinav Sunil on 5/16/23.
//

import Foundation


struct MessageRepo {
    static func load(auth: String, id: String) async throws -> [Message] {
       // sleep(1) 
        let url = URL(string: "https://cse118.com/api/v2/channel/\(id)/message")!
        var request = URLRequest(url: url)
        request.setValue("application/json", forHTTPHeaderField: "Accept") // Set the Accept header
        request.setValue("Bearer \(auth)", forHTTPHeaderField: "Authorization") // Set the Authorization header

        let (data, _) = try await URLSession.shared.data(for: request) // Use the prepared request
        
//        if let jsonString = String(data: data, encoding: .utf8) {
//            print(jsonString)
//        }
        
            return try JSONDecoder.javaScriptISO8601().decode([Message].self, from: data)
        
      //  return try JSONDecoder().decode([Message].self, from: data)
    }
    
    
    static func add(auth: String, id: String, content: String) async throws {
        let url = URL(string: "https://cse118.com/api/v2/channel/\(id)/message")!
        var request = URLRequest(url: url)
        let Content = ["content": content]
        let jsonData = try JSONEncoder().encode(Content)
        
        
        request.setValue("application/json", forHTTPHeaderField: "Accept") // Set the Accept header
        request.setValue("Bearer \(auth)", forHTTPHeaderField: "Authorization")
        request.httpMethod = "POST"
        request.httpBody = jsonData
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        
        
        
        
        let (data, _) = try await URLSession.shared.data(for: request) // Use the prepared request
        
//        if let jsonString = String(data: data, encoding: .utf8) {
//            print(jsonString)
//        }
        
        
        //let session = URLSession.shared
       // let task = session.dataTask(with: request) // ChatGPT
        //task.resume() //ChatGPT
    }
    
    static func delete(auth: String, id: String) async throws {
        let url = URL(string: "https://cse118.com/api/v2/message/\(id)")!
        var request = URLRequest(url: url)
        
        
        request.setValue("application/json", forHTTPHeaderField: "Accept") // Set the Accept header
        request.setValue("Bearer \(auth)", forHTTPHeaderField: "Authorization")
        request.httpMethod = "DELETE"
        
        let (data, _) = try await URLSession.shared.data(for: request)
        
    }
    
    
}
